byzcoin/viewchange/viewchange.go

```diff
@@ -159,11 +158,11 @@ func (c *Controller) Start(myID network.ServerIdentityID, genesis skipchain.Skip
 					ctr = c.processAnomaly(reqNew, &meta, ctr)
 				}
 			}
-			if meta.countOf(ctr) > 2*f && meta.stateOf(ctr) < startedTimerState && meta.acceptOf(ctr) {
+			if meta.countOf(ctr) >= 2*f && meta.stateOf(ctr) < startedTimerState && meta.acceptOf(ctr) {
 				// To avoid starting the next view-change too
 				// soon, start view-change timer after
-				// receiving 2*f+1 view-change messages.
-				timer.Reset(time.Duration(math.Pow(2, float64(ctr))) * initialDuration)
+				// receiving 2*f view-change messages.
+				timer.Reset(initialDuration * 2)
```

Comment:
i'm not too sure about this change. in section 4.5.2 of the PBFT paper, the time need to increase exponentially. this is because we're working in a partially synchronous network where we don't know the real network latency. the paper also says we need to wait for 2f+1 messages before starting the timer.

in my opinion, if the problem we're trying to solve is allow a 5-node cluster to operate when 2 nodes are down (at least try to make sure that the 3 nodes are not stuck), then I don't think this is the right fix. because this scenario violates the PBFT assumption and it's expected that the remaining nodes will be suck or get into an inconsistent state.

Reply:
The best approach I can think of is to upper bound the expiration timer. Because it's unrealistic for computer networks to have a delay of 5+ minutes. Alternatively you can write a bcadmin command to reset the viewchange expiration timer.

Reply:
That's right for 2f+1, I misread a paragraph. The upper bound solution is fine I guess.

Reply:
I always thought it's `2f+2` for ByzCoin, no?

Reply:
the view change is following this paper: http://pmg.csail.mit.edu/papers/osdi99.pdf which states `2f+1`

Reply:
Thanks - I was not aware that the view-change has another threshold than ByzCoin...